File: Two pointers/leetcode_977.cpp

// Problem link ===>> https://leetcode.com/problems/squares-of-a-sorted-array/description/

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

class Solution
{
public:
    vector<int> sortedSquares(vector<int> &nums)
    {
        int n = nums.size();
        int l = 0, r = n - 1;

        vector<int> v;
        while (1)
        {
            if (nums[l] * nums[l] > nums[r] * nums[r])
            {
                v.push_back(nums[l] * nums[l]);
                l++;
            }
            else
            {
                v.push_back(nums[r] * nums[r]);
                r--;
            }

            if (l == r)
            {
                v.push_back(nums[l] * nums[l]);
                break;
            }

            reverse(v.begin(), v.end());

            for (int x : v)
            {
                cout << x << " ";
            }
            cout << endl;
        }
    }
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;
    }
    return 0;
}

File: Two pointers/leetcode_977.py

# Problem link ===>>  https://leetcode.com/problems/squares-of-a-sorted-array/description/
# submission link == =>> https://leetcode.com/problems/squares-of-a-sorted-array/submissions/1592677336/

from typing import List


class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l, r = 0, n - 1
        v = []

        while l <= r:
            if abs(nums[l]) > abs(nums[r]):
                v.append(nums[l] * nums[l])
                l += 1
            else:
                v.append(nums[r] * nums[r])
                r -= 1

        return v[::-1]


if __name__ == "__main__":
    solution = Solution()
